code/modules/mob/dead/observer/observer.dm

```diff
@@ -296,7 +296,11 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 	set name = "Ghost"
 	set desc = "Relinquish your life and enter the land of the dead."
 
-	if(stat == DEAD)
+	if(src.health < 0 && src.health > -95.0) //crit people
+		succumb()
+		sleep(5)
+		ghostize(1)
+	else if(stat == DEAD)
 		ghostize(1)
 	else
 		var/response = alert(src, "Are you -sure- you want to ghost?\n(You are alive. If you ghost, you won't be able to play this round for another 30 minutes! You can't change your mind so choose wisely!)","Are you sure you want to ghost?","Ghost","Stay in body")
```

Comment:
Due to how else blocks work, yes, this should never be reached if it reaches the code you added in first.
Don't make a workaround, figure out what's happening to cause this instead.